### PR TITLE
Adds docs

### PR DESCRIPTION
**Pull request by morris25**

### CODE REVIEW

File: src/Example.jl [hunk 1]
```diff
@@ -1,7 +1,18 @@
 module Example
 export hello, domath
 
+"""
+    hello(who::String)
+
+Returns "Hello, `who`"
```
Comment (nalimilan) [inline src/Example.jl]: The manual recommends using the imperative form instead:
```suggestion
Return "Hello, `who`".
```
Same below.

**Comment by nalimilan** [comment]
Thanks. Could you also add the code needed to deploy the docs to GitHub to .travis.yml?

File: src/Example.jl (hunk 1 above)
Comment (ararslan) [reply src/Example.jl]: This wasn't actually fixed.

Comment (morris25) [reply src/Example.jl]: Whoops, sorry I was dealing with something else and apparently didn't actually read the comment.

**Comment by morris25** [comment]
Thanks for catching my mistakes. I must've been more distracted than I thought that day.

**Comment by KristofferC** [comment]
We should probably use the Documenter.jl recommended way of doing this https://juliadocs.github.io/Documenter.jl/stable/man/hosting/#.travis.yml-Configuration-1.

**Comment by morris25** [comment]
I'll note that [adding the ssh keys](https://juliadocs.github.io/Documenter.jl/stable/man/hosting/#SSH-Deploy-Keys-1) will probably have to be done by one of the package owners for the deployment to work.

**Comment by ararslan** [comment]
Done.

File: docs/make.jl [hunk 2]
```diff
@@ -0,0 +1,9 @@
+using Documenter, Example
+
+makedocs(modules = [Example], sitename = "Example.jl")
+
+if get(ENV, "TEST_DOCS", "false") != "true"
```
Comment (fredrikekre) [inline docs/make.jl]: Remove this check?

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
 *.jl.cov
 *.jl.*.cov
 *.jl.mem
+/docs/build/
```
Comment (fredrikekre) [inline .gitignore]: Maybe also add `/docs/Manifest.toml` to this list.

**Comment by ararslan** [comment]
> Can you enable Travis on this repo too?

Done. It looks like it used to be enabled but got turned off at some point.

**Comment by ararslan** [comment]
It shares the JuliaLang Travis queue so the builds will not be fast.

**Comment by fredrikekre** [comment]
Maybe disable 0.7 and nightly and only run 0.6 and 1.0?

**Comment by ararslan** [comment]
That would certainly be kinder on CI, but this is intended to be an example for others to craft their packages, so restricting the CI builds because it's annoying in the JuliaLang org seems unfortunate.

**Comment by morris25** [comment]
Is there anything else that needs to be fixed on this?

**Comment by cecileane** [comment]
The badge to the [stable](https://julialang.github.io/Example.jl/stable) doc gives a 404 "file not found" error, probably because there hasn't been a release since the addition of docs built by Documenter. Making a new release could fix this issue.

**Comment by nalimilan** [comment]
Good catch, I've tagged a new version.

**Pull request merged**